Assignment4Test.java

```diff
@@ -0,0 +1,195 @@
+import org.junit.jupiter.api.Test;
+
+import static com.elifguler.Assignment4.findNumberOfIslandsDisjointSet;
+import static com.elifguler.Assignment4.findNumberOfIslandsRecursive;
+import static org.junit.jupiter.api.Assertions.*;
+
+class Assignment4Test {
+    @Test
+    void findNumberOfIslandsRecursiveTestNormal() {
+        boolean[][] map = {
```

Comment:
Good tests!

To save you some typing/copy-and-pasting you could have these maps as `private static final boolean[][] MAP_4x4_3_ISLANDS = ...` etc in `Assignment4Test`, and just reuse them in tests for recursive or disjoing set implementations.